File: src/application/strategies/factories/abstracts/interface_application_factory.py
from abc import ABC, abstractmethod
from application.strategies.abstracts.base_application_strategy import BaseApplicationStrategy
from domain.abstracts.base_domain import BaseDomain

class IApplicationFactory(ABC):
    @abstractmethod
    def get_strategy(
        self, name: str | type[BaseDomain] | None
    ) -> BaseApplicationStrategy:
        pass

File: angular-app/src/app/core/application/strategies/factories/abstracts/interface-application-factory.ts
import { BaseDomain } from '../../../../domain/abstracts/base-domain';
import { BaseApplicationStrategy, DomainConstructor } from '../../abstracts/base-application-strategy';

export interface InterfaceApplicationFactory {
  getStrategy(name: string | DomainConstructor<BaseDomain> | null): BaseApplicationStrategy;
}
